spinalcordtoolbox/reports/qc2.py

```diff
@@ -984,3 +1016,30 @@ def assign_label_colors_by_groups(labels):
                 color_list[label - labels.min()] = sampled_colors[j % len(sampled_colors)]
 
     return color_list
+
+
+def crop_with_mask(array, img_crop, pad=3):
+    first_slice = min(np.where(img_crop.data)[0]) - pad
+    last_slice = max(np.where(img_crop.data)[0]) + pad
+    return array[first_slice:last_slice]
```

Comment:
Right now, the syntax `array[first_slice:last_slice]` will only ever crop the first dimension. Since we reorient to SAL for QC, I think this means we're only cropping the SI axis? 

But maybe we should make this clearer by adding a parameter and docstring:

```suggestion
def crop_with_mask(array, img_crop, pad=3, axis=0):
    """
    Crop array along a specific axis based on nonzero slices in the reference image.
    
    Note: We use axis=0 by default because QC images are reoriented to SAL, therefore we
          crop the SI axis by default.
    """
    first_slice = min(np.where(img_crop.data)[axis]) - pad
    last_slice = max(np.where(img_crop.data)[axis]) + pad
    idx = [slice(None)] * array.ndim            # Start with full image
    idx[axis] = slice(first_slice, last_slice)  # Limit axis to slice range
    return array[idx]
```

We can use this to crop both the SI axis and/or the LR axis (for sagittal images) by changing the value of `axis`.

Reply:
Also, I think there might be a few bugs here:

- Due to the pad value, it's possible for first/last slice to go outside the array bounds.
- I think there might be an off-by-one error, since the range `slice(start, stop)` doesn't actually include `stop`. So, I think we need to use `stop + 1`.

```suggestion
def crop_with_mask(array, img_crop, pad=3, axis=0):
    """
    Crop array along a specific axis based on nonzero slices in the reference image.
    
    Note: We use axis=0 by default because QC images are reoriented to SAL, therefore we
          crop the SI axis by default.
    """
    # get extents of segmentation used for cropping
    first_slice = min(np.where(img_crop.data)[axis])
    last_slice = max(np.where(img_crop.data)[axis])
    # pad (but make sure the indices are within the image bounds)
    start = max(first_slice - pad, 0)
    stop = min(last_slice + pad, img_crop.data.shape[axis])
    # crop the image at the specified axis
    idx = [slice(None)] * array.ndim   # Start with full image
    idx[axis] = slice(start, stop + 1) # Limit axis to slice range
    return array[idx]
```